ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -47,9 +54,109 @@ CompactionDetails getCompactionDetails(BlobStore blobStore) throws StoreExceptio
     if (usedCapacity >= (storeConfig.storeMinUsedCapacityToTriggerCompactionInPercentage / 100.0) * totalCapacity) {
       List<String> potentialLogSegments = blobStore.getLogSegmentsNotInJournal();
       if (potentialLogSegments != null) {
-        details = new CompactionDetails(time.milliseconds() - messageRetentionTimeInMs, potentialLogSegments);
+        // fetch valid data size from BlobStoreStats
+        BlobStoreStats blobStoreStats = blobStore.getBlobStoreStats();
+        Pair<Long, NavigableMap<String, Long>> validDataPerLogSegments = blobStoreStats.getValidDataSizeByLogSegment(
+            new TimeRange(time.milliseconds() - messageRetentionTimeInMs, 0));
```

Comment:
yes, for now, all stats requests are determined on demand by the BlobStoreStats. So, once we have bucketing, we can revisit this.